app/controllers/sessions_controller.rb

```diff
@@ -19,7 +17,20 @@ def destroy
     redirect_to root_url
   end
 
-  def load_user
-    @user = User.find_by email: params[:session][:email].downcase
-  end
+  private
+
+    def load_user
+      @user = User.find_by email: params[:session][:email].downcase
```

Comment:
downcase và downcase! khác nhau gì em

Reply:
Chố này theo em là không dùng được downcase! .downcase! là nó chuyển chuỗi về ký tự thường và gán luôn cho chính đó. Trường hợp truyền vào 1 email đã in thường sẵn tức sẽ không có thay đổi nào thì nó trả về nil. Như vậy nó insert vào data trường email là nil luôn.
Nên em không có dùng email.downcase! là vì lý do đó. Mà không biết em nghĩ thế đúng hay sai? Có gì anh giải thích thêm giúp em ạ.

Reply:
Chỗ này theo em là không dùng được downcase! .downcase! là nó chuyển chuỗi về ký tự thường và gán luôn cho chính đó. Trường hợp truyền vào 1 email đã in thường sẵn tức sẽ không có thay đổi nào thì nó trả về nil. Như vậy nó insert vào data trường email là nil luôn.
Nên em không có dùng email.downcase! là vì lý do đó. Mà không biết em nghĩ thế đúng hay sai? Có gì anh giải thích thêm giúp em ạ.

Reply:
@hhhhuy112 Có sai không anh?A trả lời giúp em cái này ạ? Thank a